src/iron.rs

```diff
@@ -67,32 +67,40 @@ impl Default for Timeouts {
 
 /// Protocol used to serve content. Future versions of Iron may add new protocols
 /// to this enum. Thus you should not exhaustively match on its variants.
-#[derive(Clone)]
+#[derive(Copy, Clone)]
 pub enum Protocol {
     /// Plaintext HTTP/1
     Http,
     /// HTTP/1 over SSL/TLS
-    #[cfg(feature = "ssl")]
-    Https {
-        /// Path to SSL certificate file
-        certificate: PathBuf,
-        /// Path to SSL private key file
-        key: PathBuf
-    }
+    Https,
+    #[doc(hidden)]
+    __Extensible,
```

Comment:
To allow adding new variants in the future.

I'd actually recommend making the enum part private, and expose function constructors instead.

```rust
pub struct Protocol(Proto);
enum Proto { .. }

impl Protocol {
    pub fn http() -> Protocol {
        Protocol(Proto::Http)
    }
}
```